src/xcode/ENA/ENA/Source/Scenes/ENSetting/ExposureNotificationSettingViewController.swift

```diff
@@ -206,64 +207,78 @@ extension ExposureNotificationSettingViewController {
-					lastActionCell = cell
-				}
-			case .euTracingCell:
-				let euTracingCell = tableView.dequeueReusableCell(withIdentifier: ENSettingModel.Content.euTracingCell.cellType.rawValue, for: indexPath)
-				if let euTracingCell = euTracingCell as? EuTracingTableViewCell {
-					euTracingCell.configure()
-					return euTracingCell
-				}
-			case .tracingCell, .actionDetailCell:
-				switch enState {
-				case .enabled, .disabled:
-					let tracingCell = tableView.dequeueReusableCell(withIdentifier: ENSettingModel.Content.tracingCell.cellType.rawValue, for: indexPath)
-					if let tracingCell = tracingCell as? TracingHistoryTableViewCell {
-						let colorConfig: (UIColor, UIColor) = (self.enState == .enabled) ?
-							(UIColor.enaColor(for: .tint), UIColor.enaColor(for: .hairline)) :
-							(UIColor.enaColor(for: .textPrimary2), UIColor.enaColor(for: .hairline))
-						let activeTracing = store.tracingStatusHistory.activeTracing()
-						let text = [
-							activeTracing.exposureDetectionActiveTracingSectionTextParagraph0,
-							activeTracing.exposureDetectionActiveTracingSectionTextParagraph1]
-							.joined(separator: "\n\n")
-
-						let numberOfDaysWithActiveTracing = activeTracing.inDays
-						let title = NSLocalizedString("ExposureDetection_ActiveTracingSection_Title", comment: "")
-						let subtitle = NSLocalizedString("ExposureDetection_ActiveTracingSection_Subtitle", comment: "")
-
-						tracingCell.configure(
-							progress: CGFloat(numberOfDaysWithActiveTracing),
-							title: title,
-							subtitle: subtitle,
-							text: text,
-							colorConfigurationTuple: colorConfig
-						)
-						return tracingCell
-					}
-				case .bluetoothOff, .restricted, .notAuthorized, .unknown, .notActiveApp:
-					if let cell = cell as? ActionCell {
-						cell.configure(for: enState, delegate: self)
-					}
-				}
-			case .descriptionCell:
-				cell.configure(for: enState)
+
+		guard let cell = tableView.dequeueReusableCell(withIdentifier: content.cellType.rawValue, for: indexPath) as? ConfigurableENSettingCell else {
+			return UITableViewCell()
+		}
+
+		switch content {
+		case .banner:
+			cell.configure(for: enState)
+		case .actionCell:
+			if let lastActionCell = lastActionCell {
+				return lastActionCell
 			}
-			return cell
-		} else {
+			if let cell = cell as? ActionCell {
+				cell.configure(for: enState, delegate: self)
+				lastActionCell = cell
+			}
+		case .euTracingCell:
+			return euTracingCell(for: indexPath, in: tableView)
+		case .tracingCell, .actionDetailCell:
+			switch enState {
+			case .enabled, .disabled:
+				return tracingCell(for: indexPath, in: tableView)
+			case .bluetoothOff, .restricted, .notAuthorized, .unknown, .notActiveApp:
+				(cell as? ActionCell)?.configure(for: enState, delegate: self)
+			}
+		case .descriptionCell:
+			cell.configure(for: enState)
+		}
+
+		return cell
+	}
+
+	private func euTracingCell(for indexPath: IndexPath, in tableView: UITableView) -> UITableViewCell {
```

Comment:
Since this has been moved to private methods, I think it would be good to use the MARKs

Reply:
How shall we use MARKS to structure a file with
1 protocol, 1 class, and 6 extensions? Any suggestions?

Reply:
Basically the question arises why all this is in one file.
Within the class I would set the MARKs and then of course distribute them in the extensions.
Superfluous marks can then be removed.
Keep the protocol on top - I guess we will refactor most of them later.

Reply:
I added a private MARK for now, but this class needs an additional structural cleanup later